src/doc/src/reference/lints.md

```diff
@@ -0,0 +1,115 @@
+# Lints
+
+Note: [Cargo's linting system is unstable](unstable.md#lintscargo) and can only be used on nightly toolchains
+## Allowed-by-default
+
+These lints are all set to the 'allow' level by default.
+- [`implicit_features`](#implicit_features)
+
+### `implicit_features`
+
+Note: This only runs on edition 2021 and below
+
+#### What it does
+Checks for implicit features for optional dependencies
+
+#### Why it is bad
+By default, cargo will treat any optional dependency as a [feature]. As of
+cargo 1.60, these can be disabled by declaring a feature that activates the
+optional dependency as `dep:<name>` (see [RFC #3143]).
+
+In the 2024 edition, `cargo` will stop exposing optional dependencies as
+features implicitly, requiring users to add `foo = ["dep:foo"]` if they
+still want it exposed.
+
+For more information, see [RFC #3491]
+
+#### Example
+```toml
+edition = "2021"
+
+[dependencies]
+bar = { version = "0.1.0", optional = true }
+
+[features]
+# No explicit feature activation for `bar`
+```
+
+Instead, the dependency should have an explicit feature:
+```toml
+edition = "2021"
+
+[dependencies]
+bar = { version = "0.1.0", optional = true }
+
+[features]
+bar = ["dep:bar"]
+```
+
+[feature]: https://doc.rust-lang.org/cargo/reference/features.html
+[RFC #3143]: https://rust-lang.github.io/rfcs/3143-cargo-weak-namespaced-features.html
+[RFC #3491]: https://rust-lang.github.io/rfcs/3491-remove-implicit-features.html
+
+
+## Warn-by-default
+
+These lints are all set to the 'warn' level by default.
+- [`unknown_lints`](#unknown_lints)
+- [`unused_optional_dependency`](#unused_optional_dependency)
+
```

Comment:
I went with all the lint levels at the top, and then all the lints sorted by name. I went this route since we do not have many lints currently, and we can always move to separate files in the future